Python/Product/TestAdapter/TestExecutor.cs

```diff
@@ -85,6 +81,52 @@ class TestExecutor : ITestExecutor {
             RunTestCases(receiver.Tests, runContext, frameworkHandle);
         }
 
+        private Dictionary<string, PythonProjectSettings> GetSourceToSettings(IRunSettings settings) {
+            var doc = new XPathDocument(new StringReader(settings.SettingsXml));
+            XPathNodeIterator nodes = doc.CreateNavigator().Select("/RunSettings/Python/TestCases/Project");
+            Dictionary<string, PythonProjectSettings> res = new Dictionary<string, PythonProjectSettings>();
+
+            foreach (XPathNavigator project in nodes) {
+                PythonProjectSettings projSettings = new PythonProjectSettings(
+                    project.GetAttribute("path", ""),
+                    project.GetAttribute("home", ""),
+                    project.GetAttribute("workingDir", ""),
+                    project.GetAttribute("interpreter", ""),
+                    project.GetAttribute("pathEnv", ""),
+                    ParseBoolAttribute(project, "nativeDebugging")
+                );
+
+                foreach (XPathNavigator environment in project.Select("Environment/Variable")) {
+                    projSettings.Environment[environment.GetAttribute("name", "")] = environment.GetAttribute("value", "");
+                }
+
+                string djangoSettings = project.GetAttribute("djangoSettingsModule", "");
+                if (!String.IsNullOrWhiteSpace(djangoSettings)) {
+                    projSettings.Environment["DJANGO_SETTINGS_MODULE"] = djangoSettings;
+                }
+
+                foreach (XPathNavigator searchPath in project.Select("SearchPaths/Search")) {
+                    projSettings.SearchPath.Add(searchPath.GetAttribute("value", ""));
+                }
+
+                foreach (XPathNavigator test in project.Select("Test")) {
+                    res[test.GetAttribute("file", "")] = projSettings;
```

Comment:
And some fallback behaviour - crash if we need to, but with something that makes it obvious that an assertion failed in the release build..

---
In reply to: [67421586](https://github.com/Microsoft/PTVS/pull/1337#discussion_r67421586) [](ancestors = 67421586,67386339)